### PR TITLE
Make NPCs better with (player's) meds

### LINKED ISSUE #17117: NPC Medical Aid Limited
Major Issues

 - They wont take antibodies outside of complete trust
 - They cant recieve antiseptic powder or liquid disinfectant
 - they wont take off splints when their wounds heal

Minor Issues

  - They wont take pain pills
  - Still wont drink booze
 

 - you can to 'trick' them into buffs by trading passive artifacts to them
 - Not all artifact powers work, and ones like teleport do while dexup does not.
 -  If you mod in extended effect armor (such as a wearable stimpack) they can be tricked into putting it on

### PR DESCRIPTION
**Pull request by Coolthulhu**
- `"USE_ON_NPC"` flag obsoleted, now all heal actors should be working
- Added ability to use liquid meds on NPCs (notably disinfectant)
- `"NPC_SAFE"` flag will make NPCs skip the trust check
- NPCs who have some time to spare will look at their splints. If they don't cover any broken parts, they'll swap the splint to the other side. If that doesn't help, they will take the splint off

Fixes #17117

Also a seemingly small, but notable balancing change: got rid of the ability to mend wounds with cotton balls pulled out from pillows. They can still be used for bleeding, but not hp damage.

### CODE REVIEW

File: src/player.cpp [hunk 1]
```diff
@@ -9289,6 +9289,44 @@ int player::drink_from_hands(item& water) {
...
+        amount_used = the_med->type->invoke( this, the_med, pos );
+        if( amount_used <= 0 ) {
+            return false;
+        }
+    }
+
+    // @todo Get the target it was used on
+    // Otherwise injecting someone will give us addictions etc.
+    consume_effects( *the_med );
+    moves -= 250;
```
Comment (illi-kun) [inline src/player.cpp]: Is it possible to use ```mod_moves( -250 )``` here?

File: src/player.h [hunk 2]
```diff
@@ -710,6 +710,12 @@ class player : public Character, public JsonSerializer, public JsonDeserializer
         /** Used for eating a particular item that doesn't need to be in inventory.
          *  Returns true if the item is to be removed (doesn't remove). */
         bool consume_item( item &eat );
+        /**
+         * Consumes an item as medication.
+         * Can be used to heal others (using `pos` argument), if med type handles that.
+         * Will complain if the item isn't actually a med type.
+         */
+        bool consume_med( item &target, const tripoint &pos );
```
Comment (mugling) [inline src/player.h]: Document the return type
Use `doxygen` syntax (eg. `@param` and `@returns`)

File: data/json/items/comestibles/med.json [hunk 3]
```diff
@@ -39,10 +38,11 @@
         "price" : 14500,
         "charges" : 10,
         "stack_size" : 200,
+        "flags" : "NPC_SAFE",
```
Comment (mugling) [inline data/json/items/comestibles/med.json]: JSON style for these files is consistently `flags` as last field

File: src/item.cpp [hunk 4]
```diff
@@ -3373,6 +3375,20 @@ bool item::is_food() const
     return type->comestible != nullptr;
 }
 
+bool item::is_med() const
```
Comment (mugling) [inline src/item.cpp]: `is_medicine` or `is_medication`
Preferably the latter as we should eventually migrate to a `MEDICATION

File: src/item.h [hunk 5]
```diff
@@ -739,6 +739,8 @@ class item : public JsonSerializer, public JsonDeserializer, public visitable<it
  bool is_food() const;                // Ignoring the ability to eat batteries, etc.
  bool is_food_container() const;      // Ignoring the ability to eat batteries, etc.
  bool is_ammo_container() const; // does this item contain ammo? (excludes magazines)
+    bool is_med() const;            // Is it a medication that only pretends to be food?
```
Comment (mugling) [inline src/item.h]: Avoid inconsistent indentation. The entire block can be updated later

File: src/player.cpp [hunk 6]
```diff
@@ -9289,6 +9289,44 @@ int player::drink_from_hands(item& water) {
...
+        the_med = &target.contents.front();
+    } else if( target.is_med() ) {
+        the_med = &target;
+    } else {
+        debugmsg( "%s tried to use a %s as medication", name.c_str(), target.tname().c_str() );
+        return false;
+    }
+
+    auto req_tool = item::find_type( the_med->type->comestible->tool );
+    if( req_tool->tool ) {
```
Comment (mugling) [inline src/player.cpp]: If it's a tool you should use the `ammo_*` family of functions

Comment (Coolthulhu) [inline src/player.cpp]: Doesn't `ammo_*` family require an actual item? We're operating on a type here.

**Pull request merged**